src/components/proxy.rs

```diff
@@ -18,6 +18,9 @@ pub(crate) mod error;
 pub mod packet_router;
 mod sessions;
 
+#[cfg(target_os = "linux")]
+pub mod xdp;
```

Comment:
Also minor nit: Can we move this under `net` as opposed to in the proxy. I think it belongs there.